### PR TITLE
Add support for Custom Class Colors

### LINKED ISSUE #2450: Supporting !ClassColors
**Is your feature request related to a problem? Please describe.**
I searched up a way to change Shaman Class Color to the normal Blue in >>Classic WoW<< . I Found the AddOn " !ClassColors "
and its working greatly and other Big AddOns like ElvUI do support it. It doesnt taint the UI. In WeakAuras however the Classcolor given for shamans still stay pink.

**Describe the solution you'd like**
I'd love to see WeakAuras Supporting that AddOn so Shamans are blue.

**Describe alternatives you've considered**
I asked on the WeakAura-Discord on how to maybe fix that myself. And "Infus" told me I should open up a ticket and here I am.

**Additional context**

In the Screenshot below you see what I basically just described. Shamans Color is blue on all Elements except that WeakAura above the targetframe. 
bluepinkidk

here its the clock wa
bluepinkidk2

Thanks for taking your time. :)

### PR DESCRIPTION
**Pull request by Stanzilla**
# Description

Adds support for [ClassColors](https://www.curseforge.com/wow/addons/classcolors)

Fixes #2450 

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Test A
- [ ] Test B

# Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

**Comment by mrbuds** [comment]
It miss changes to WeakAuras.class_types and WeakAuras.class_color_types, and you need to replace GetClassColoredTextForUnit

**Comment by Stanzilla** [comment]
> It miss changes to WeakAuras.class_types and WeakAuras.class_color_types, and you need to replace GetClassColoredTextForUnit

We have a comment there that the types are unused but yes about GetClassColoredTextForUnit

**Comment by mrbuds** [comment]
You could implement the callback to update colors CUSTOM_CLASS_COLORS:RegisterCallback(method[, handler])

**Comment by Stanzilla** [comment]
> You could implement the callback to update colors CUSTOM_CLASS_COLORS:RegisterCallback(method[, handler])

I don't think we need to since we're not caching the table for long term use. We query it every time anyway.

**Comment by mrbuds** [comment]
> We have a comment there that the types are unused 

That comment is partially outdated, class_types is used in Prototypes.lua for class load conditions

File: WeakAuras/Types.lua [hunk 1]
```diff
@@ -9,7 +9,16 @@ local LSM = LibStub("LibSharedMedia-3.0");
 local wipe, tinsert = wipe, tinsert
 local GetNumShapeshiftForms, GetShapeshiftFormInfo = GetNumShapeshiftForms, GetShapeshiftFormInfo
 local GetNumSpecializationsForClassID, GetSpecializationInfoForClassID = GetNumSpecializationsForClassID, GetSpecializationInfoForClassID
-local WrapTextInColorCode, GetClassColor = WrapTextInColorCode, GetClassColor -- for Classic
+local WrapTextInColorCode = WrapTextInColorCode
+
+local function WA_GetClassColor(classFilename)
```
Comment (InfusOnWoW) [inline WeakAuras/Types.lua:14]: It seems only the fourth return value is used?

**Comment by InfusOnWoW** [comment]
Didn't Buds request it to be merged to master instead?

**Comment by Stanzilla** [comment]
Yeah I'll rebase it once the review is done, did not want to clutter that up

**Review by InfusOnWoW — APPROVED** [review]

**Pull request merged**